tests/reduction/reduction_common.h

```diff
@@ -0,0 +1,147 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides common code for reduction tests
+//
+*******************************************************************************/
+
+#ifndef __SYCL_CTS_TEST_REDUCTION_COMMON_H
+#define __SYCL_CTS_TEST_REDUCTION_COMMON_H
+
+#include "../common/common.h"
+#include "../common/type_coverage.h"
+
+namespace reduction_common {
+
+static constexpr size_t number_iterations{10};
+
+static sycl::range<1> range{number_iterations};
+static sycl::nd_range<1> nd_range{range, range};
+
+static const auto scalar_types =
+    named_type_pack<char, signed char, unsigned char, short int,
+                    unsigned short int, int, unsigned int, long int,
+                    unsigned long int, float, long long int,
+                    unsigned long long int>(
+        {"char", "signed char", "unsigned char", "short int",
+         "unsigned short int", "int", "unsigned int", "long int",
+         "unsigned long int", "float", "long long int",
+         "unsigned long long int"});
+
+/** @brief Returns expected value for testing
+ *  @tparam VariableT The type of the variable with which the test runs
+ *  @tparam FunctorT The type of the functor with which the test runs
+ *  @tparam BufferT The type of the buffer with which the test runs
+ *  @param functor The functor (plus, multiplies e.t.c) with which the test runs
+ *  @param buffer The buffer that will used in parallel_for() function
+ *  @param value_for_initialization The value for initialization output variable
+ *  @retval Expected value after test will running
+ */
+template <typename VariableT, typename FunctorT, typename BufferT>
+VariableT get_expected_value(FunctorT functor, BufferT& buffer,
+                             VariableT value_for_initialization) {
+  VariableT expected_value{value_for_initialization};
+  auto buf_accessor =
+      buffer.template get_access<sycl::access_mode::read>(range);
+  for (int i{}; i < number_iterations; i++) {
+    expected_value = functor(buf_accessor[i], expected_value);
+  }
+  return expected_value;
+}
+
+/** @brief Filling buffer for using it in parallel_for and for calculate
+ * expected value
```

Comment:
Curious indent.

Reply:
Curious indent was fixed in this commit: ea039f0583fc00dfa57c74375de05863dae900d8